src/components/ExpandButtonComponent.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import Modal from '@material-ui/core/Modal';
+import IconButton from '@material-ui/core/IconButton';
+import OpenWith from '@material-ui/icons/OpenWith';
+import ComponentToTestExpandButton from './ComponentToTestExpandButton';
+
+
+function getModalStyle() {
+  const top = 50;
+  const left = 50;
+
+  return {
+    top: `${top}%`,
+    left: `${left}%`,
+    transform: `translate(-${top}%, -${left}%)`,
+  };
+}
+
+const useStyles = makeStyles((theme) => ({
+  paper: {
+    position: 'absolute',
+    width: '50%',
+    height: '80%',
+    backgroundColor: theme.palette.background.paper,
+    border: '2px solid #000',
+    boxShadow: theme.shadows[5],
+    padding: theme.spacing(2, 4, 3),
+  },
+}));
+
+interface ExpandButtonComponentInterface {
+  componentToExpand: JSX.Element;
+}
+
+export default function ExpandButtonComponent(props: ExpandButtonComponentInterface) {
+  console.log(props.componentToExpand);
+
+  const classes = useStyles();
+  const [modalStyle] = React.useState(getModalStyle);
```

Comment:
A mogłabyś powiedzieć dlaczego używasz tutaj `useState`?

Reply:
Masz rację - jest to tutaj zbędne. :) Przepisałam (bezmyślnie) - mój błąd.